core/lib/config/src/configs/chain.rs

```diff
@@ -52,6 +60,31 @@ pub enum L1BatchCommitDataGeneratorMode {
     Validium,
 }
 
+// The cases are extracted from the `PubdataPricingMode` enum in the L1 contracts,
+// And knowing that, in Ethereum, the response is the index of the enum case.
+// If the Uint token is 0 I want the Rollup case,
+// If the Uint token is 1 I want the Validium case,
+// Else, an error.
+impl Detokenize for L1BatchCommitDataGeneratorMode {
+    fn from_tokens(tokens: Vec<ethabi::Token>) -> Result<Self, Web3ContractError> {
+        let error = Err(Web3ContractError::Api(Web3ApiError::Decoder(
+            "L1BatchCommitDataGeneratorMode::from_tokens".to_string(),
+        )));
```

Comment:
Tip: It's usually helpful to include information that would help to debug the error if it ever happens, e.g. in this case the tokens you couldn't parse. And to avoid doing extra work in an `Ok(..)` cases, you can make it a function, e.g.

```rust
fn from_tokens(tokens: Vec<ethabi::Token>) -> Result<Self, Web3ContractError> {
  fn error(tokens: &[ethabi::Token]) -> Web3ContractError {
    Web3ContractError::Api(Web3ApiError::Decoder(
            format!("L1BatchCommitDataGeneratorMode::from_tokens: {:?}", tokens)
        ))
  }
  
 match tokens.as_slice() {
    // ...
    _ => Err(error(&tokens))
 }
}
```